### ISSUE TITLE
"NYC" and metro areas

### ISSUE THREAD

**Issue by thisisaaronland** [whosonfirst/whosonfirst-data#238]
Attention @nvkelso @riordan @trescube @dphiffer 

The short version is: The locality of New York City (85977539) is parented by the local admin or borough of Brooklyn (404521211).

The problem is: This is neither correct nor incorrect. There are a series of overlapping issues:

1. The polygon for NYC which is broadly correct for the 5 boroughs isn't really a "locality"
2. The label centroid for that polygon, as derived by mapshaper is in fact in Brooklyn so the PIP code is correct in saying the centroid is in Brooklyn
3. https://whosonfirst.mapzen.com/spelunker/id/404522731/ <— the local admin / borough of Manhattan
4. The part where NYC should really have multiple hierarchies – namely all 5 boroughs – or:
5. The part where NYC should probably not be classified as a locality at all, but rather a metro area
6. The part where people refer to “NYC” as only manhattan or all five boroughs depending on your view of the world
7. The part where people refer to cities and metroareas more or less interchangeably and often with diametrically opposite meanings depending on whether they are a local or a tourist

My inclination would be to finally pull the trigger on "metroareas" and:

1. Supersede 85977539 (NYC, the locality) with a new record for NYC, the metroarea
2. Parent the 5 boroughs (localadmins) with the new record
3. Merge most or all of the exonyms for NYC and Manhattan (there aren't any yet)
4. Update all the things to include the metro area - this is a bit daunting but it won't be the last time we'll need to do this so...

Immediate concerns:

* Will this make the Pelias build sad because it doesn't know about “metroareas” ?

Discuss.

**Comment by dphiffer** [comment]
In no particular order:
* Manhattan equated with NYC feels so wrong, although I understand that wrongness may strike visitors as misplaced (i.e., when the tourist types in "NYC" they may not actually *mean* all 5 boroughs). Even so, I would be really sad if this is how the data described the city.
* I might actually prefer 5 hierarchies to a metroarea, if only because elsewhere "New York Metropolitan Area" tends to include other nearby cities. It's messier though.
* Basically I like how Ingrid describes it in her Twitter bio: "A Small Island Near America"

**Comment by dphiffer** [comment]
Or maybe it's "a set of small islands." Did I just equate Manhattan with NYC?

**Comment by nvkelso** [comment]
I think of `metropolitan` (and `micropolitan` & etc) regions as **overlays**, not part of an admin structure. I propose they be added, but only listed in a place's _hierarchy_ and not be available for direct parenting (as in `parent_id`). 

Proposal:

* Add new WOF "New York metro" feature with `metroregion` placetype.  Parented by NY state, but with multiple hierarchies including New Jersey and Connecticut. 
* Locality of `New York City` (85977539) is kept around, and is still parented by NY state, and has NY state in it's hierarchy. But the new `metroregion` reference is added to the record's hierarchy. The `county` and `localadmin` parts of the locality's hierarchy should be removed. Instead those are children.
* I think the 5 boroughs should (also) be `localadmin` that are parented by the locality of New York City, with their correct local names. 
* We still need the 5 boroughs as counties though (for `county` completeness in the USA and with their correct and different names), but mark them as not relevant for label construction. They should be parented by NY city, but have NY state in their region hierarchy. 

We've talked about adding a "seat of government" flag to indicate which `locality` is the capital of what `region`, `country`, & etc. We could do the same for the new `metroregion`.

For **San Francisco**, where it's a combined county-city, what would the rule of thumb be? Keep the county but mark it as not interesting for label generation?

I'm open to `metroarea` versus `metroregion`, but whatever we use needs to accommodate the "micro" and "rural" variants e.g.: `microregion` and `ruralregion` or `microarea` and `ruralarea`.

**Comment by nvkelso** [comment]
Haha, "A Small Island Near America" reminds me of @burritojustice's [Islands of SF](http://burritojustice.com/islands-of-sf/).

![islands-of-san-francisco-2013](https://cloud.githubusercontent.com/assets/853051/14125092/32e8d14a-f5be-11e5-9ca3-67c132376083.jpg)

**Comment by dphiffer** [comment]
Awesome, also that BART map is pretty sweet!

**Comment by burritojustice** [comment]
ferry routing ftw

**Comment by nvkelso** [comment]
First Lyft car pools, next "ferry pools".

**Comment by thisisaaronland** [comment]
I can live with `metroregion` mostly on the grounds that `microregion` sounds terrible. I also do _not_ want to get involved in other meta placetypes as part of this issue.

I agree that it should be an "overlay" rather than a strict/direct ancestor. I don't think we should refer to them as "overlays" though since that really only makes sense in a cartographic context.

It sounds like we should make each of the five boroughs a locality (rather than a localadmin) and make NYC a localadmin (rather than a locality).

Having an interchangeable relationship / order of associaton between localities and localadmins is only going to perpetuate the problem that NYC (and London, to be fair) forces on things.

**Comment by burritojustice** [comment]
Uber actually had a boat service in Sydney a while back

**Comment by thisisaaronland** [comment]
Okay, so New York gets "boroughs". This is going to take (more than 30 minutes) because it gets bit finnicky, in the details.

Currently blocked on issue #239

**Comment by thisisaaronland** [comment]
Things that need to be superceded (and re-created as boroughs) :

* https://whosonfirst.mapzen.com/spelunker/id/404521211/ (Brooklyn)
* https://whosonfirst.mapzen.com/spelunker/id/404521559/ (Queens)
* https://whosonfirst.mapzen.com/spelunker/id/404522731/ (Manhattan)
* https://whosonfirst.mapzen.com/spelunker/id/404521159/ (Bronx)
* https://whosonfirst.mapzen.com/spelunker/id/404523089/ (Staten Island)

**Comment by thisisaaronland** [comment]
NYC needs to be updated to:

* Be parented by NY state (85977539)
* Having multiple hierarchies (each of the 5 counties that are co-existent with the 5 boroughs)
* Not have any local admins as ancestors

https://whosonfirst.mapzen.com/spelunker/id/85977539/

**Comment by thisisaaronland** [comment]
There are ~170K records that need to be updated and/or re-reverse-geocoded to reflect their boroughness:

https://whosonfirst.mapzen.com/spelunker/id/85977539/descendants/

**Comment by nvkelso** [comment]
+1

**Comment by thisisaaronland** [comment]
Attention @dphiffer – you and I will need to coordinate once the first part of the updates (for not-venue) things are complete so that the NY copy of BI and all it's data get updated in sync.

**Comment by thisisaaronland** [comment]
See also: whosonfirst-toolbox/scripts/issue-238-descendants.py

**Comment by thisisaaronland** [comment]
A dry-run for updating venues with sanity-checking of parent IDs in whosonfirst-toolbox/scripts/issue-238-venues.py yields (83) venues that would have their `wof:parent_id` most of which pre-date any work on Boundary Issues so this feels like an acceptable rate of error / things that will need to be (re) touched by a human, per https://github.com/whosonfirst/whosonfirst-www-boundaryissues/issues/85.

**Comment by thisisaaronland** [comment]
![screen shot 2016-03-31 at 14 18 55](https://cloud.githubusercontent.com/assets/12658759/14191240/a971b2d0-f74b-11e5-9c1f-5379205638d2.png)

**Comment by riordan** [comment]
YEAAAH

On 3/31/16 17:20, Aaron Straup Cope wrote:
> screen shot 2016-03-31 at 14 18 55
> <https://cloud.githubusercontent.com/assets/12658759/14191240/a971b2d0-f74b-11e5-9c1f-5379205638d2.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/whosonfirst/whosonfirst-data/issues/238#issuecomment-204134218>
>

-- 
David Riordan | Product Manager - Search | Mapzen
dave.riordan@mapzen.com | gpg 235D 9DC9 5EF6 277C
@riordan | +1 (203)-521-1222
https://mapzen.com | @mapzen

**Comment by thisisaaronland** [comment]
Weird, this returns no results but I am guessing it's an indexing thing in ES...

https://whosonfirst.mapzen.com/spelunker/id/85977539/descendants/?&placetype=borough

**Comment by thisisaaronland** [comment]
Still to do: Merge venues repo `issue-238` branch and update bi-1 accordingly. I think the only real issue is whether or not ops needs to be asked to include boroughs in the list of bundles they pull for the pip server...

**Comment by nvkelso** [comment]
https://github.com/mapzen/vector-datasource/issues/654

On Thu, Mar 31, 2016 at 2:45 PM, Aaron Straup Cope <notifications@github.com
